### ISSUE TITLE
Changing tab name has led to data loss of the selection within the media picker

### ISSUE THREAD

**Issue by Jeavon** [KevinJump/uSync3.0#160]
Whilst adding a load of new properties a existing tab was renamed, this causes the property to be deleted and re-created which of course leads to loss of the existing data.

In the attached screen shot the tab named "Info" was renamed "BM View" the logo property didn't change.

![image](https://user-images.githubusercontent.com/1829920/32889841-4f534090-cac4-11e7-8545-86ccdbeedc56.png)

**Comment by KevinJump** [comment]
:( 
In theory it shouldn't actually delete the property as they are matched by an internal key value (but i will check that!) so the tab doesn't matter. 

_The change might well say delete because it is a comparison of XML files it doesn't match the internal process completely :(_

**Comment by Jeavon** [comment]
The result was slightly odd because the umbraco.config and IPublishedContent contained the values but the Content Service and therefore the Umbraco UI didn't have any values so for some reason it does seem that internally the property was dropped and recreated...

**Comment by KevinJump** [comment]
yeah i need to do the tests - it shouldn't happen - but that doesn't mean it didn't ;) 

are you all latest umbraco / usync ? (just for me to recreate it)

**Comment by Jeavon** [comment]
Yup we are

```
  <package id="uSync" version="4.0.1" targetFramework="net462" />
  <package id="uSync.Core" version="6.0.1" targetFramework="net462" />
  <package id="UmbracoCms" version="7.7.5" targetFramework="net462" />
```

Thanks Kevin!

**Comment by KevinJump** [comment]
Right i am close :) 

I think there has been a change in behaviour in the core - but it's my bad for relying on it

the lookup in theory is fine, but before we get there we create a new tab and delete the old one - I am failry sure that in older umbraco's this just put all of the values in the tab into 'generic' and then the next bit of the process which updates the properties would put them back in. 

looks like deleting a tab now deletes the properties in the tab :( 

bit of a logic re-write coming up (not massive though).

**Comment by Jeavon** [comment]
Phew, very glad you've tracked it down!

**Comment by KevinJump** [comment]
Hi 
fixed - and released uSync 4.0.2 (Core 6.0.2) - just waiting for nuget to index it

**Comment by Jeavon** [comment]
Completely awesome Kevin!

Many thanks!
